articles/intro-to-C++-part01/index.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: intro-to-C++-part01
+title: Introduction to C++
+description: This introduction is about C++ and goes through topics such as why C++?, who uses C++in the industry, basic implementation.
+author: Parampreet Singh
+date: 2020-08-17T00:00:00-12:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/intro-to-C++-part01/hero.jpg
+    alt: image example
+---
+There are many languages being used in the industry at the moment, but C++ has been used for a long time now (been around since 1979). It is being used by companies from young startups to well-established companies like Facebook and Google.
+<!--more-->
+### Why C++?
+C++ is a powerful language. It allows a developer to have a lot of control as to how they use their computer resources. It is considered remarkably cheap and fast. Therefore, Developers always prefer C++ to build web browsers, game engines, games, and desktop apps.
+
+Some of the best applications built with C++ are:
+
+- Apple OS X Operating System
+- Microsoft Operating System including Microsoft Office, Internet Explorer, and Visual Studio
+- MySQL Server
+- Mozilla Firefox
+- Adobe System
+- Google Chrome
+- PayPal
+- Amazon
+
+And the list goes on. You can count the applications built using C++ for days.
+
+C++ is a beginner-friendly language that anybody can start learning with or without any prior programming experience. The online community is so strong that you will never feel left out of any resources or lack any help regarding C++.
+
+In this series of guides, will we go from basics to the more advanced levels.
+So let's jump straight into learning.
+
+### What do you need beforehand
+- A working PC or a Computer
+- An IDE
+
+The first question that comes to mind would be, what is an IDE? It is an Integrated Development Environment where you will write your C++ code and run it.
+To get started, you can download [Visual studio code](https://code.visualstudio.com/) as it is a simple and beginner-friendly text editor software.
+
+Let's dive into coding:
+
+### Hello World
+Let's write a Hello World program. We will break it down line by line:
+
+1. // This is a sample C++ Program
+2. #include<iostream>
+3. using namespace std;
+4. int main() {
+5. cout<<"Hello World"<<endl;
+6. return 0;
+7. }
+
+
+This program will print Hello World to the screen. Let's examine it line by line to see what it means.
+
+- Line 1 - **//This is a sample C++ Program**
+
+Any line of a program that begins with a "//" is a comment. The compiler will ignore this line.
+
+Comments are there to be read by people. You should always write comments for each major step of the algorithm to explain something weird or cool that is happening. That way, anyone working along with you or for you would be able to understand what you were trying to do.
```

Comment:
'something weird or cool that is happening' is subjective and not very helpful in guiding people on how/when to use comments. Provide specific examples here.